crates/testing/src/task_helpers.rs

```diff
@@ -148,15 +235,67 @@ async fn build_quorum_proposal_and_signature(
         timestamp: 0,
         proposer_id: *api.public_key(),
     };
-    let signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
+
+    let mut signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
         .expect("Failed to sign leaf commitment!");
-    let proposal = QuorumProposal::<TestTypes> {
-        block_header,
-        view_number: ViewNumber::new(view),
+    let mut proposal = QuorumProposal::<TestTypes> {
+        block_header: block_header.clone(),
+        view_number: ViewNumber::new(1),
         justify_qc: QuorumCertificate::genesis(),
         timeout_certificate: None,
         proposer_id: leaf.proposer_id,
     };
+    for cur_view in 2..=view {
+        consensus.state_map.insert(
+            ViewNumber::new(1),
+            View {
+                view_inner: ViewInner::Leaf {
+                    leaf: leaf.commit(),
```

Comment:
I'm a bit confused about why the leaf is the same each time we insert it into the state map. Shouldn't it change for each view, or is this a simplification implementation for testing?

Reply:
+1, also do we even need to store the anything in the state_map or the leaves at all?

Reply:
Looks like we won't retrieve anything from the `state_map` or `saved_leaves` so there's no need to update them here.

Reply:
`leaf` will be re-assigned at the end of the loop every time QAQ, maybe the naming is causing some confusion here, should I change it to `cur_leaf`?

Reply:
Oh sorry, I didn't realize this `leaf` was mutable. The renaming sounds good.

However, I'm still confused about why we need to do the `saved_leaves` and `state_map` insertions. In lines 205 and 211, don't we already have the parent state and leaf corresponding to the requested view? Why do we need to build and store them again?

Reply:
Without updating `state_map`, the test will get an error during `visit_leaf_ancestors`. For `saved_leaves`, see the comment above.

Reply:
Hmm, I still don't fully get it. In lines 205 and 211, we already have the parent state and leaf corresponding to the _high QC_ (sorry in my previous reply I said it's to the _requested view_, but it's actually high QC). If we want to update `state_map` and `saved_leaves`, I think it should start from the view after the high QC, instead of view 1. Otherwise, are we assuming the parent view is the genesis view?

Reply:
Aha, sorry I didn't see your earlier comment about this. For line 205 and 211, yeah they're reading the parent state (can I call this view 0? maybe not well-defined but I need a way to separate it from view 1), then we create the leaf for view 1 in line 228, and in the loop we start to create leaves for view 2 and higher. 

And, to validate a proposal at `view x`, we'll need valid `state_map` and `saved_leaves` for `view x-1`. I'm not sure why when `x = 1`, it already had valid `state_map`(I guess due to some initialization), but starting `x = 2`, we do need to save these states for `x - 1` manually to pass all the checks, that's also why I keep these updates at the start of the loop.

Does all these make sense? Let me know if it raised any concerns :)